core/domain/exp_services.py

```diff
@@ -792,6 +793,15 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
                 'which is too old. Please reload the page and try again.'
                 % (exploration_model.version, exploration.version))
 
+    if feconf.ENABLE_STRING_CLASSIFIER:
```

Comment:
@seanlip However, one part of the logic happens before the version increment on the Exploration domain object, the second part happens after the increment. I'm not sure how we could work around that. The version increment has to happen between both these separate chunks of the logic.